Mathlib/CategoryTheory/Sites/OneHypercover.lean

```diff
@@ -139,6 +139,220 @@ def multifork (F : Cᵒᵖ ⥤ A) :
     dsimp
     simp only [← F.map_comp, ← op_comp, E.w])
 
+section Category
+
+/-- A morphism of pre-`1`-hypercovers of `S` is a family of refinement morphisms commuting
+with the structure morphisms of `E` and `F`. -/
+@[ext]
+structure Hom (E F : PreOneHypercover S) where
+  /-- The map between indexing types of the coverings of `S` -/
+  s₀ (i : E.I₀) : F.I₀
+  /-- The map between indexing types of the coverings of the fibre products over `S`. -/
+  s₁ {i j : E.I₀} (k : E.I₁ i j) : F.I₁ (s₀ i) (s₀ j)
+  /-- The refinement morphisms between objects in the coverings of `S`. -/
+  h₀ (i : E.I₀) : E.X i ⟶ F.X (s₀ i)
+  /-- The refinement morphisms between objects in the coverings of the fibre products over `S`. -/
+  h₁ {i j : E.I₀} (k : E.I₁ i j) : E.Y k ⟶ F.Y (s₁ k)
+  w₀ (i : E.I₀) : h₀ i ≫ F.f (s₀ i) = E.f i := by aesop_cat
+  w₁₁ {i j : E.I₀} (k : E.I₁ i j) : h₁ k ≫ F.p₁ (s₁ k) = E.p₁ k ≫ h₀ i := by aesop_cat
+  w₁₂ {i j : E.I₀} (k : E.I₁ i j) : h₁ k ≫ F.p₂ (s₁ k) = E.p₂ k ≫ h₀ j := by aesop_cat
+
+attribute [reassoc (attr := simp)] Hom.w₀
+attribute [reassoc] Hom.w₁₁ Hom.w₁₂
+
+/-- The identity refinement of a pre-`1`-hypercover. -/
+@[simps]
+def Hom.id (E : PreOneHypercover S) : Hom E E where
+  s₀ := _root_.id
+  s₁ := _root_.id
+  h₀ _ := 𝟙 _
+  h₁ _ := 𝟙 _
+
+variable {E : PreOneHypercover.{w} S} {F : PreOneHypercover.{w'} S}
+  {G : PreOneHypercover S}
+
+/-- Composition of refinement morphisms of pre-`1`-hypercovers. -/
+@[simps]
+def Hom.comp (f : E.Hom F) (g : F.Hom G) : E.Hom G where
+  s₀ := g.s₀ ∘ f.s₀
+  s₁ := g.s₁ ∘ f.s₁
+  h₀ i := f.h₀ i ≫ g.h₀ _
+  h₁ i := f.h₁ i ≫ g.h₁ _
+  w₁₁ := by simp [w₁₁, w₁₁_assoc]
+  w₁₂ := by simp [w₁₂, w₁₂_assoc]
+
+/-- The induced index map `E.I₁' → F.I₁'` from a refinement morphism `E ⟶ F`. -/
+@[simp]
+def Hom.s₁' (f : E.Hom F) (k : E.I₁') : F.I₁' :=
+  ⟨⟨f.s₀ k.1.1, f.s₀ k.1.2⟩, f.s₁ k.2⟩
+
+instance category : Category (PreOneHypercover S) where
```

Comment:
By making it an unnamed instance, we can generate the simp lemmas for identities and compositions automatically.
```suggestion
@[simps! id_s₀ id_s₁ id_h₀ comp_h₁ comp_s₀ comp_s₁ comp_h₀ comp_h₁]
instance : Category (PreOneHypercover S) where
```
Below, it would be better to use `E ⟶ F` instead of `E.Hom F`.